src/openrct2-ui/scripting/CustomWindow.cpp

```diff
@@ -1436,6 +1437,32 @@ namespace OpenRCT2::Ui::Windows
         }
     }
 
+    void CloseWindowsOwnedByPlugin(std::shared_ptr<Plugin> plugin)
+    {
+        // Get all the windows that need closing
+        std::vector<std::shared_ptr<rct_window>> customWindows;
+        for (const auto& window : g_window_list)
+        {
+            if (window->classification == WC_CUSTOM)
+            {
+                auto customWindow = reinterpret_cast<CustomWindow*>(window.get());
+                auto customInfo = reinterpret_cast<CustomWindowInfo*>(customWindow->custom_info);
```

Comment:
Do we really need `reinterpret_cast`, couldn't they be `dynamic_cast`?

Reply:
That would be unnecessary inefficient, the class check will give us confidence that the type is correct.